synapse/storage/databases/main/user_directory.py

```diff
@@ -1238,7 +1238,13 @@ def _parse_query_postgres(search_term: str) -> Tuple[str, str, str]:
     search_term = _filter_text_for_index(search_term)
 
     escaped_words = []
-    for word in _parse_words(search_term):
+    for index, word in enumerate(_parse_words(search_term)):
+        if index > 10:
```

Comment:
> We "fix" this by cheating and only searching for the first ten words.

If we care about this being accurate

```suggestion
        if index > 9:
```

Reply:
haha, oops

Reply:
```suggestion
        if index >= 10:
```